### DISCUSSION TITLE
Dynamic taxes

### DISCUSSION THREAD

**Discussion by karolkielecki** [saleor/saleor#10258] [Announcements]
Attention Saleors! 📢 

Welcome to the first phase of our new approach to taxes. Dynamic taxes has landed in our main branch! Thanks to the new synchronous tax webhooks, you can delegate tax calculations to Saleor Apps 🙌

Learn more ⤵️
https://docs.saleor.io/docs/3.x/developer/extending/apps/synchronous-webhooks/tax-webhooks

**Comment by karolkielecki** [comment]
PS Dynamic taxes will be available in the upcoming stable 3.6 release.

**Comment by sixthgear** [comment]
Hi @karolkielecki! I left this comment initally on https://github.com/saleor/saleor-docs/pull/476#issuecomment-1209795745, but I just found this discussion here so thought I'd copy here.

I'm a user currently testing this functionality and I have a couple questions. Please forgive me if this isn't the appropriate place, but I'm not sure where else on the web to ask such things (besides opening a new Issue).

Is there (or will there be) a provision through this new taxes webhook to:

- Provide a name or code for the applied tax so it can be shown to the checkout
- Provide multiple tax rates, so that each tax can be itemized individually on the checkout. Eg. in Canada we have GST + PST in some provinces, but a single HST (harmonized) in other provinces

Additionally, while on the topic of synchronous webhooks, is there anything on the roadmap pertaining to:

- A surcharge (or fee or levy) datatype which can be applied to products and itemized in the cart/checkout.
- If not, the ability to control checkout contents via synchronous webhook. For instance, when a product is added or removed, the app validates the checkout lines and appends the appropriate fee (configured as a saleor product)
- Again, if not, can you tell me if the plugin system is a more appropriate place to configure this type of logic? I've avoided diving into plugins so far because I wanted to deploy a clean docker instance, but would make an exception to solve this particular problem.

Thank you so much for your work on Saleor. I'm excited for the 3.6 release and seeing Saleor become more and more capable with each update.

**Reply by karolkielecki** [reply]
Hi @sixthgear, first of all -- thanks for your kind words! My colleague has replied to your [comment](https://github.com/saleor/saleor-docs/pull/476#issuecomment-1210211936) here. Please let me know if that answers your questions. And, if you find any issues/bugs or have more questions, please let us know :)
